Description: No SIGPWR on FreeBSD
 Removes warning and definition for SIGPWR on FreeBSD as it doesn't
 exist there anyhow.
Author: Jon Boden <jon@ubuntubsd.org>
Bug-Debian: https://bugs.debian.org/832148
Reviewed-by: Craig Small <csmall@debian.org>
Last-Update: 2016-08-28
--- a/proc/sig.c
+++ b/proc/sig.c
@@ -56,7 +56,7 @@
 #endif
 
 /* It seems the SPARC libc does not know the kernel supports SIGPWR. */
-#ifndef SIGPWR
+#if defined(__linux__) && !defined(SIGPWR)
 #  warning Your header files lack SIGPWR. (assuming it is number 29)
 #  define SIGPWR 29
 #endif
@@ -84,7 +84,9 @@
   {"PIPE",   SIGPIPE},
   {"POLL",   SIGPOLL},  /* IO */
   {"PROF",   SIGPROF},
+#ifdef SIGPWR
   {"PWR",    SIGPWR},
+#endif
   {"QUIT",   SIGQUIT},
   {"SEGV",   SIGSEGV},
 #ifdef SIGSTKFLT
